### DISCUSSION TITLE
Conditional generation?

### DISCUSSION THREAD

**Discussion by Vitor-Almeida** [lk-geimfari/mimesis#1246] [Q&A] [answered]
Hello!

I'm generating random "product_id" (integers) from 1 to 1000, and for every product, I also want to generate a "sales_price", but I need the sales price to stay kind of constant between the same “product_id”.

for example:
product_id | sales_price
213 | $12,11
151 | $2,41
213 | $12,03
154 | $103,45

I can’t have a large variance in “sales_price” between the same products in different rows.
I was thinking in setting a specific price range to every possible “product_id”,
is this possible using the library? How would I achieve this?

One more question, is there any function on the library that lets me implement a numeric value that follow a gaussian distribution or any other distribution? Is there any available example using custom providers?

Thank you!

**Comment by lk-geimfari** [comment, marked as answer]
We don't have such functionality out of the box, unfortunately.


You can read about custom providers here: https://mimesis.name/en/master/getting_started.html#custom-providers
